packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,688 @@
+    `nimble-mapping-mention-user`. For example, if the client application provide the duplicate `key` values that stores the user ID, it will be an
+    issue in scenarios when parsing the mention user from just a user ID to username.
+
+_Events_
+
+-   `mention-update` - This event fires when the `@` character added to the editor and for every character input after `@`.
+    This fires with the `eventData` containing the current text that is added after the `@` character.
+
+    This can be achieved through Tiptap's `onUpdate()` and `onStart()` methods in `render` function in
+    [suggestion](https://tiptap.dev/api/utilities/suggestion#render) configurations. This event fires with the `eventData` containing the
+    current text that is added after the `@` character.
+
+    This event will be triggered in the following scenarios to perform filtering in the client application. The below scenarios will also fire
+    when there is a configuration element `nimble-rich-text-mention-users` even without the mapping elements:
+
+    1. When a user inserts the character (e.g., `@`) into the editor, which activates the mention popup.
+    2. When a user adds or removes text after inserting the mention character into the editor.
+    3. When a user repositions the cursor between the text segments added after the mention character.
+
+    Refer the [accessibility](#accessibility) section to know more details about when it requires to emit the event for performing the filtering in the client application.
+
+    Conversely, this event will not be fired when:
+
+    1. A user moves the cursor away either using mouse or any key down events that moves the cursor from current position to outside of `@`
+       and characters after that with less than two white spaces while the popup is open.
+    2. A user selects an option from the dropdown list either using mouse click/pressing Enter or Tab.
+    3. A user adds two or more spaces after triggering the mention popup.
+    4. In the absence of `nimble-rich-text-mention-users` and no mapping elements.
+
+_Methods_
+
+-   `checkValidity()` - this returns `true` if the configuration of the `@mention` mapping data is valid and `false` otherwise.
+
+#### User mapping element (Non-visible configuration element):
+
+This mapping element is employed to establish a connection between the value displayed in the mapping view and the corresponding value stored within
+the markdown string. For instance, the username for an `@mention` is contained in the `display-name` attribute, which is used for display in the
+mention view, while the user ID URL is contained in the `mention-url` attribute which is used to store within the markdown string.
+
+_Component Name_
+
+-   `nimble-mapping-mention-user`
+
+_Props/Attrs_
+
+-   `mention-url`: string
+-   `display-name`: string
+
+#### User mention view (Visible UI element):
```

Comment:
I have captured the expected copying and pasting behavior of "@mention" from viewer to editor and also from editor to editor in the HLD. 

In the current prototype, we are facing some style issues that make the glitchy behavior you mentioned. Also, rendering a shadow root component like `nimble-rich-text-mention-view` within a `contenteditable` div causes quite a lot of basic behavioral issues compared to the normal elements. This is also a reason that caused us to not render the [`nimble-anchor` within a content editable element](https://github.com/ni/nimble/issues/1516).  I also couldn't find more details about rendering a shadow root element and we doubt that it might be due to some styling that is encapsulated within the shadow root. So we thought we would try to achieve the same behavior I mentioned in the HLD which is also the same as [Tiptap's `@mention` behavior](https://tiptap.dev/api/nodes/mention#usage) in the time of actual implementation.